packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/api/sitemap.ts

```diff
@@ -19,24 +19,40 @@ const sitemapApi = async (
     siteName: config.jssAppName,
   });
 
-  const sitemapPath = await sitemapXmlService.getSitemap(id as string);
+  if (id) {
+    const sitemapPath = await sitemapXmlService.getSitemap(id as string);
 
-  if (sitemapPath) {
-    const isAbsoluteUrl = sitemapPath.match(ABSOLUTE_URL_REGEXP);
-    const sitemapUrl = isAbsoluteUrl ? sitemapPath : `${config.sitecoreApiHost}${sitemapPath}`;
-    res.setHeader('Content-Type', 'text/xml;charset=utf-8');
+    if (sitemapPath) {
+      const isAbsoluteUrl = sitemapPath.match(ABSOLUTE_URL_REGEXP);
+      const sitemapUrl = isAbsoluteUrl ? sitemapPath : `${config.sitecoreApiHost}${sitemapPath}`;
+      res.setHeader('Content-Type', 'text/xml;charset=utf-8');
 
-    return new AxiosDataFetcher()
-      .get(sitemapUrl, {
-        responseType: 'stream',
-      })
-      .then((response: AxiosResponse) => {
-        response.data.pipe(res);
-      })
-      .catch(() => res.redirect('/404'));
-  }
+      return new AxiosDataFetcher()
+        .get(sitemapUrl, {
+          responseType: 'stream',
+        })
+        .then((response: AxiosResponse) => {
+          response.data.pipe(res);
+        })
+        .catch(() => res.redirect('/404'));
+    }
+  } else {
+    const sitemaps = await sitemapXmlService.fetchSitemaps();
+    const SitemapLinks = sitemaps
```

Comment:
Do we need to handle case where `sitemaps` is empty?

Reply:
When we try to open sitemap.xml link without sitemap-1.xml

Reply:
I discussed with Alan and we decided if sitemaps is empty the script will redirect to 404 page